015-exploratory-test-module.md

```diff
@@ -0,0 +1,163 @@
+
+#### CreativeExploratory
+
+Creative exploration consists of several steps:
+1. start exploration application
+2. (interaction with app) the user has to provide YAML, which he wants to apply
+3. (interaction with app) yes/no (is component running?)
+4. if there is another component that the user wants to apply to go to `3rd step` else go to next step 
+5. transcript with wrong and success paths (reporting) = user end the end of testing will have a report about his work
+
+### Reporting
+
+In the last phase of testing user will have a transcript of all his decisions with YAMLs. For clarity, here is one of the transcript:
+
+```
+[
+    [
+        echo "apiVersion: apps/v1
+              kind: Deployment
+              metadata:
+                name: strimzi-cluster-operator
+                labels:
+                  app: strimzi
+              spec:
+                replicas: 1
+               ..." 
+        | 
+        oc apply -f -        
+    ],
+    [
+        echo "apiVersion: kafka.strimzi.io/v1beta1
+             kind: Kafka
+             metadata:
+               name: my-cluster
+             spec:
+               kafka:
+                 version: 2.6.0
+                 replicas: 3
+             ..."
+        | 
+        oc apply -f -
+    ],
+    [
+        echo "apiVersion: kafka.strimzi.io/v1beta1
+                kind: KafkaUser
+                metadata:
+                  name: my-user
+                  labels:
+                    strimzi.io/cluster: my-cluster
+                spec:
+                ..."
+        | 
+        oc apply -f -
+    ]
+    ...
+```
+
+As you can see it is the list of commands, which is needed to do to reproduce the same behaviour that the user had. 
+In the plain text it will look like this:
+
+```
+======
+1.step
+======
+echo "apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: strimzi-cluster-operator
+        labels:
+          app: strimzi
+      spec:
+        replicas: 1
+       ..." 
+| 
+oc apply -f -  
+======
+2.step
+======
+echo "apiVersion: kafka.strimzi.io/v1beta1
+     kind: Kafka
+     metadata:
+       name: my-cluster
+     spec:
+       kafka:
+         version: 2.6.0
+         replicas: 3
+     ..."
+| 
+oc apply -f -
+======
+3.step
+======
+echo "apiVersion: kafka.strimzi.io/v1beta1
+        kind: KafkaUser
+        metadata:
+          name: my-user
+          labels:
+            strimzi.io/cluster: my-cluster
+        spec:
+        ..."
+| 
+oc apply -f -
+```
+
+## Proposal
+
+This proposal suggests to:
+* Create new repository `exploratory-test` in `strimzi`
+
+##### Advantages:
+* reduce time when manually testing some components
+* pre-cursor for marathon testing
+* gives us report (which combination were buggy)
+* everybody can play with it and find bugs
+##### Dis-advantages:
+* It may lead to the fact that the scenarios will not be original and sometimes it will be tested redundantly
+* the user has to input syntactically valid YAMLs
```

Comment:
YAMLs could be easily verified in the code. However, the overall Kafka yaml have to be correct (I point to use-case when user will put the only spec of Kafka for example)